xoto3/dynamodb/write_versioned/run.py

```diff
@@ -70,41 +38,47 @@ def versioned_transact_write_items(
     item_version_attribute: str = "item_version",
     last_written_attribute: str = "last_written_at",
 ) -> VersionedTransaction:
-    """Performs a read-transact-write loop over multiple items across
-    multiple tables until there are no interruptions causing the
-    transaction to fail, or until the transaction expires (as
-    specified by the attempts_iterator or the default timed_retry).
-
-    Allows PUTs and DELETEs - an UPDATE with versioning is no different
-    from a PUT.
-
-    You do not have to write back to every item specified as part of
-    the transaction, but we will assert the unchanged state of every
+    """Performs a read-transact-write loop over multiple versioned
+    puts/deletes across multiple tables, until there are no
+    interruptions causing the transaction to fail, or until the
+    transaction expires (as specified by the attempts_iterator or the
+    default timed_retry).
+
+    Your transaction _must_ be a pure function, as it will likely be
```

Comment:
This wording is a bit more strict than the previous one I commented on.

Reply:
yeah, it is. The truth, of course, that you might be able to get away with making it not a truly pure function. But that gets into questions of idempotency and lots of other stuff that in general are best left alone.